src/source-updater.js

```diff
@@ -47,13 +47,54 @@ export default class SourceUpdater {
     this.mediaSource = mediaSource;
     this.processedAppend_ = false;
 
+    this.bufferItemsList_ = [];
+
     if (mediaSource.readyState === 'closed') {
       mediaSource.addEventListener('sourceopen', createSourceBuffer);
     } else {
       createSourceBuffer();
     }
   }
 
+  appendToBufferInfoQueue_(timestampOffset, duration, byteLength) {
```

Comment:
Instead of putting all of the buffer info in `SourceUpdater` and keeping track of it, maybe it makes sense for us to add byte info to the segment metadata track https://github.com/videojs/videojs-contrib-hls#segment-metadata and we can make use of that.

Reply:
For reference, the PR to include the required information in the segment metadata is #1210.